File: Summer/Main/system_energy.py
from spinenergy import *

def system_energy_1(sx,sy,m,n,jv,qv):
	
	r = 0.0

	for i in range(m):
		for j in range(n):
			r += spinenergy1(sx,sy,i,j,jv,qv,m,n)

	r = r/2.0

	return r

def system_energy_p(sx,sy,sz,m,n,p,jv,qv):

	r = 0.0

	for i in range(m):
		for j in range(n):
			for k in range(p):
				r += spinenergy_p(sx,sy,sz,i,j,k,jv,qv,m,n,p)

	r = r/2.0

	return r

File: Semester 7/Main/system_energy.m
function r = system_energy(jv,x,y,z)

r = 0;

[m,n] = size(x);

for i = 1:m
    for j = 1:n
        r = r + spin_energy(i,j,jv,x,y,z);
    end
end

r = r/2;
end